doc/designs/disable-stale-users.md

```diff
@@ -0,0 +1,273 @@
+
+There is always the possibility that a long-inactive user account will be
+disabled (nsAccountLock'd) by DSU even if that user managed to logon the
+previous few seconds before or during the DSU window due to:
+* the time it takes for the timestamp attribute to be replicated to the DSU
+runner
+* the time it takes for nsAccountLock to be replicated from the DSU runner
+to the replica this user uses.
+
+There is no way to prevent that from happening with the present design, but
+the probability of that happening is low.
+
+
+## Example
+
+### Environments with mandatory, periodic password changes
+
+Given the following requirements and facts:
+* stale accounts must be disabled after 90 days' inactivity
+* grace period is decided to be 7 days
+* the tool runs every day
+* replication is expected to complete in 2h maximum
+
+setting the maximum password lifetime to 80 days or less would meet the
+guidelines as 80+7+1 (give or take 2h) is less than 90.
+
+### Environments with no mandatory password changes
+
+If the requirement is to disable users not seen for 90 days maximum, updating
+the attribute every 35 days is good enough.
+At worst, the user would have authenticated the first day and at least once
+during this 35 day period, say day #29. The attribute would therefore contain
+the date from the first day, as day #29 would have been considered too close to
+the existing timestamp to warrant updating it. The user account would then be
+disabled after 61 days of inactivity which fits (loosely) the requirement.
+In short, having a 90-day inactive user disabling requirement would by default
+disable accounts inactive for 55 (90-35) to 90 days.
+
+
+## Limitations
+
+### Environments with mandatory, periodic password changes
+
+If regulations call for stale/inactive accounts to be disabled within X
+number of days, then the lifetime of the passwords for these accounts must be
+set to a value lower than:
+* X,
+* minus the expected grace period mentioned above,
+* minus the maximum amount of time between two invocations of the DSU tool,
+* minus the expected maximum replication delay from the replica the DSU tool is
+installed on to the farthest point of the cluster.
+
+### Environments with no mandatory password changes
+
+The longer the time period, the less update activity for the timestamp
+attribute. But using a longer period runs the risk of disabling users who have
+been gone for too short a time. For instance, setting the update period to 65
+days would disable accounts inactive for the last 25 to 90 days. Whether this
+fits the site's requirements is up to the administrators to determine.
+
+Sites requiring a better accuracy must take care not to set the update period
+too low for performance reasons. It is strongly recommended to make sure the
+update period is longer than the expected maximum PTO or simply "away from IPA"
+period for most user accounts. This is especially true when these away periods
+are in sync, like in schools.
```

Comment:
Implicitly because then all passwords will be reset within a short time frame right? Hence lots of writes which == lesser perf?

Reply:
Yes. Do you think this needs +v?

Reply:
verbose mode added.